src/plugins/data/public/index_patterns/fields/field_list.ts

```diff
@@ -45,19 +52,36 @@ export class FieldList extends Array<Field> implements IFieldList {
     this.groups.get(field.type)!.set(field.name, field);
   };
   private removeByGroup = (field: IFieldType) => this.groups.get(field.type)!.delete(field.name);
+  private fieldFormats: FieldFormatsStart;
+  private toastNotifications: ToastsStart;
 
-  constructor(indexPattern: IndexPattern, specs: FieldSpec[] = [], shortDotsEnable = false) {
+  constructor(
+    { fieldFormats, toastNotifications }: Deps,
```

Comment:
@ppisljar just to clarify. Some time ago we had the similar issue with `SearchSource`. That time we decided to move it to runtime contract instead of passing required services from each places where we use that class. 
From my point of view here we should use the same pattern.

Reply:
i think for FieldList this is fine, as its only used inside this service and its not statically exported. but for IndexPatternFieldList i agree, it seems that the getIndexPatternFieldList function that is currently inside discover should be available on data plugin start contract ?

Reply:
@ppisljar  IndexPatternFieldList === FieldList

![image](https://user-images.githubusercontent.com/20072247/80597183-290cf600-8a30-11ea-895d-93a5f7d3ade8.png)

My suggestion: 
- Remove static `IndexPatternFieldList` at all from our static contract cause it confuse and doesn't allow us simply navigate between files. 
-indexPattern runtime constact should return 
```
{
   fieldList: {
     create: () => IFieldList  
  }
}

```